cpp/example_code/aurora/getting_started_with_db_clusters.cpp

```diff
@@ -0,0 +1,1278 @@
+    if (!parameterGroupFamily.empty()) {
+        request.SetDBParameterGroupFamily(parameterGroupFamily);
+    }
+
+    Aws::RDS::Model::DescribeDBEngineVersionsOutcome outcome =
+            client.DescribeDBEngineVersions(request);
+
+    if (outcome.IsSuccess()) {
+        engineVersionsResult = outcome.GetResult().GetDBEngineVersions();
+    }
+    else {
+        std::cerr << "Error with Aurora::DescribeDBEngineVersionsRequest. "
+                  << outcome.GetError().GetMessage()
+                  << std::endl;
+    }
+
+    return outcome.IsSuccess();
+}
+// snippet-end:[cpp.example_code.aurora.DescribeDBEngineVersions]
+
+// snippet-start:[cpp.example_code.aurora.DescribeDBInstances]
+
+//! Routine which gets a DB instance description.
+/*!
+ \sa describeDBCluster()
+ \param dbInstanceIdentifier: A DB instance identifier.
+ \param instanceResult: The 'DBInstance' object containing the description.
+ \param client: 'RDSClient' instance.
+ \return bool: Successful completion.
+ */
+bool AwsDoc::Aurora::describeDBInstance(const Aws::String &dbInstanceIdentifier,
+                                        Aws::RDS::Model::DBInstance &instanceResult,
+                                        const Aws::RDS::RDSClient &client) {
+    Aws::RDS::Model::DescribeDBInstancesRequest request;
+    request.SetDBInstanceIdentifier(dbInstanceIdentifier);
+
+    Aws::RDS::Model::DescribeDBInstancesOutcome outcome =
+            client.DescribeDBInstances(request);
+
+    bool result = true;
+    if (outcome.IsSuccess()) {
+        instanceResult = outcome.GetResult().GetDBInstances()[0];
+    }
+        // This example does not log an error if the DB instance does not exist.
+        // Instead, it returns false.
+    else if (outcome.GetError().GetErrorType() !=
+             Aws::RDS::RDSErrors::D_B_INSTANCE_NOT_FOUND_FAULT) {
+        result = false;
+        std::cerr << "Error with Aurora::DescribeDBInstances. "
+                  << outcome.GetError().GetMessage()
+                  << std::endl;
+    }
+
+    return result;
+}
+// snippet-end:[cpp.example_code.aurora.DescribeDBInstances]
+
+// snippet-start:[cpp.example_code.aurora.DescribeOrderableDBInstanceOptions]
+
+//! Routine which gets available DB instance classes, displays the list
+//! to the user, and returns the user selection.
+/*!
+ \sa chooseDBInstanceClass()
+ \param engineName: The DB engine name.
+ \param engineVersion: The DB engine version.
+ \param dbInstanceClass: String for DB instance class chosen by the user.
+ \param client: 'RDSClient' instance.
+ \return bool: Successful completion.
+ */
+bool AwsDoc::Aurora::chooseDBInstanceClass(const Aws::String &engine,
+                                           const Aws::String &engineVersion,
+                                           Aws::String &dbInstanceClass,
+                                           const Aws::RDS::RDSClient &client) {
+    std::vector<Aws::String> instanceClasses;
+    Aws::String marker;
+    do {
```

Comment:
Optional, but to me it wasn't immediately obvious that the checking of the marker is in order to get multiple pages of data (in all cases for paging), so maybe consider mentioning that in a comment?